### PR TITLE
close #5792 with testcase

### LINKED ISSUE #5792: error when a discriminator branch is constructed with certain sets
```nim
type
    T = enum
        a
        b
        c
    U = object
        case k: T
        of a:
            x: int
        of {b, c} - {a}: # {b, c} does work
            y: int

static:
    echo(({b, c} - {a}).repr) # {1..2}
    echo(({b, c}).repr) # {1, 2}

discard U(k: b, y: 1) # "Error: a case selecting discriminator 'k' with value 'b' appears in the object construction, but the field(s) 'y' are in conflict with this value"
```

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: tests/sets/t5792.nim [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+type
+  T = enum
+    a
+    b
+    c
+  U = object
+    case k: T
+    of a:
+      x: int
+    of {b, c} - {a}: # {b, c} does work
```
Comment (ringabout) [inline tests/sets/t5792.nim]:
```suggestion
    of {b, c} - {a}:
```

File: tests/sets/t5792.nim [hunk 2]
```diff
@@ -0,0 +1,17 @@
+discard """
+  matrix: "--gc:refc, -gc:arc"
```
Comment (ringabout) [inline tests/sets/t5792.nim]:
```suggestion
  matrix: "--gc:refc; -gc:arc"
```

File: tests/sets/t5792.nim [hunk 3]
```diff
@@ -0,0 +1,17 @@
+discard """
+  matrix: "--gc:refc; -gc:arc"
```
Comment (ringabout) [inline tests/sets/t5792.nim]:
```suggestion
  matrix: "--gc:refc; --gc:arc"
```

File: tests/sets/t5792.nim [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    b
+    c
+  U = object
+    case k: T
+    of a:
+      x: int
+    of {b, c} - {a}:
+      y: int
+
+discard U(k: b, y: 1)
```
Comment (timotheecour) [inline tests/sets/t5792.nim:17]:
```suggestion
doAssert U(k: b, y: 1).y == 1
doAssert U(k: b, y: 1).k == b
```
to be on safe side

maye add this in followup PR?

Comment (ringabout) [reply tests/sets/t5792.nim:17]: see https://github.com/nim-lang/Nim/pull/17727

**Pull request merged**